exoplayer-codelab-01/src/main/java/com/example/exoplayer/PlayerActivity.kt

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (C) 2017 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.example.exoplayer
+
+import android.annotation.SuppressLint
+import android.os.Bundle
+import android.view.View
+import androidx.appcompat.app.AppCompatActivity
+import com.example.exoplayer.databinding.ActivityPlayerBinding
+import com.google.android.exoplayer2.MediaItem
+import com.google.android.exoplayer2.SimpleExoPlayer
+import com.google.android.exoplayer2.util.Util
+
+/**
+ * A fullscreen activity to play audio or video streams.
+ */
+class PlayerActivity : AppCompatActivity() {
+
+    private val viewBinding by lazy { ActivityPlayerBinding.inflate(layoutInflater) }
+
+    private var player: SimpleExoPlayer? = null
+
+    private var playWhenReady = true
+    private var currentWindow = 0
+    private var playbackPosition: Long = 0
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(viewBinding.root)
+    }
+
+    public override fun onStart() {
+        super.onStart()
+        if (Util.SDK_INT > 23) {
+            initializePlayer()
```

Comment:
Instead of calling these inline like this, consider creating a [`LifecycleObserver`](https://developer.android.com/topic/libraries/architecture/lifecycle). It could be initialized in `onCreate` and added as an observer to the activity with `getLifecycle().addObserver(...)`.

If I were building it, I might consider having two parameters to the observer -- the 'init' and 'release' lifecycle states. Then, in the observer, check the state and call the appropriate method. (This way there wouldn't have to be duplicate code to init/release the player in onStart/onResume and onPause/onStop)

Reply:
I considered this, however my major prompt was to keep it super simple so users can copy paste, and also to keep it as close to the original java codelab as possible as this is a first pass kotlin conversion.

After the convergence implementation, I can then explore using androidx dependencies like in lifecycle observers, and then even plausibly move the exoplayer instance to a `ViewModel`.

I can't do that at the moment as it would require extra dependencies, and even more edits to the codelab doc.

Reply:
I think this is a good suggestion but agree that we might want to do this in a follow up PR. I would file a separate issue on this repo so that this suggestion doesn't get forgotten.

Reply:
Created #65